Adyen/Validators/DateValidator.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// A generic validator for dates.
+public final class DateValidator: Validator {
+
+    private let formatter: DateFormatter
+
+    /// Initializes the date validator.
+    ///
+    /// - Parameters:
+    ///   - format: Date format.
+    ///   See  [Apple docs](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/DataFormatting/Articles/dfDateFormatting10_4.html)
+    public init(format: String) {
```

Comment:
what if we make this function accept `DateValidator.Format` instead of `String`. That way an invalid format can't be passed.
if (in the furure) we need to use regular date/time styles like `DateFormatter.Style` we can have another init that would accept those enum values.

Reply:
yep, I was also considering about it... 
🤔